config.c

```diff
@@ -13,15 +13,60 @@
 ConfigStruct config;
 static char *parse_FTLconf(FILE *fp, const char * key);
 static void release_config_memory(void);
+void getpath(FILE* fp, const char *option, const char *defaultloc, char **pointer);
 
 char *conflinebuffer = NULL;
 
+void getLogFilePath(void)
+{
+	FILE *fp;
+	char * buffer;
+
+	// Try to open default config file. Use fallback if not found
+	if( ((fp = fopen(FTLfiles.conf, "r")) == NULL) &&
+	    ((fp = fopen(FTLfiles.snapConf, "r")) == NULL) &&
+	    ((fp = fopen("pihole-FTL.conf", "r")) == NULL))
+	{
+		printf("Notice: Found no readable FTL config file");
+	}
+
+	// Read LOGFILE value if available
+	// defaults to: "/var/log/pihole-FTL.log"
+	buffer = parse_FTLconf(fp, "LOGFILE");
+
+	errno = 0;
+	// Use sscanf() to obtain filename from config file parameter only if buffer != NULL
+	if(buffer == NULL || sscanf(buffer, "%127ms", &FTLfiles.log) == 0)
```

Comment:
`sscanf` returns `EOF` on error, which is -1 (but the macro should be used instead). This may need to be used instead of zero.

Reply:
If the option is, e.g. `LOGFILE=`, we read an empty string. From the `man` page it is not entirely clear if `sscanf()` would return `0` or `EOF` is returned if the string passed to the function is of length zero. You may be right. Shall we test for `EOF` or shall we be more general an do something like ` < 1` ?

Reply:
Testing for `< 1` will probably do fine (we could also do `!= 1`).